src/org/zaproxy/zap/extension/api/CoreAPI.java

```diff
@@ -453,6 +460,22 @@ public void process(HttpMessage msg) {
 			Stats.clear(this.getParam(params, PARAM_KEY_PREFIX, ""));
 			return ApiResponseElement.OK;
 			
+		} else if (ACTION_DELETE_SITE_NODE.equals(name)) {
+			try {
+				String url = params.getString(PARAM_URL);
+				String method = getParam(params, PARAM_METHOD, "GET");
```

Comment:
Maybe it should not default to GET? We need to handle the case where there's no method being sent (i.e. thus having null value), to allow to delete directory nodes.